docs/reference/enable-gpu.rst

```diff
@@ -1,46 +1,153 @@
-Enable Nvidia GPUs
+.. _nvidia-gpu:
+
+Enable NVIDIA GPUs
 ==================
 
-To be able to run workloads on GPU with DSS, you need to have:
+The following section describes how to configure the DSS to utilize the host's
+NVIDIA GPU. This is achieved by configuring the underlying MicroK8s, that the
+DSS is relying for running the containerised workloads.
+
+Prerequisites
+^^^^^^^^^^^^^
+
+* Have run the :ref:`tutorial` tutorial
+
+Install the NVIDIA Operator
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To ensure the DSS can utilise the NVIDIA GPUs:
+
+1. The host will need to have the NVIDIA drivers installed
+2. MicroK8s will need to be setup to utilise those drivers
+
+MicroK8s is leveraging the `NVIDIA Operator`_ to for setting up and
+configuring the NVIDIA runtime. The NVIDIA Operator will also install
+the NVIDIA drivers, if they are not present already on the host machine.
+
+To enable the NVIDIA runtimes on MicroK8s, run the following command:
+
+.. code-block:: bash
+
+   sudo microk8s enable gpu
+
+.. note::
+   The NVIDIA Operator will detect if the NVIDIA drivers are not present at all
+   and will install them in this case. But the NVIDIA Operator can also detect
+   if you have drivers already installed. In this case the NVIDIA Operator will
+   use the host's drivers.
+
+Verify the NVIDIA Operator is up
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Before spinning up workloads that you need to ensure the GPU Operator has
+been successfully initialized. This can take around 5 minutes.
+
+Ensure DaemonSet is Ready
+"""""""""""""""""""""""""
+
+First we need to ensure that the DaemonSet for the Operator Validator is
+created:
+
+
+.. code-block:: bash
+
+  while ! sudo microk8s.kubectl get ds -n gpu-operator-resources nvidia-operator-validator
+  do
+    sleep 5
+  done
+
+.. note::
+   It will take some seconds for the DaemonSet to get created. The above command will
+   return a message like the following at the beginning:
+   ``Error from server (NotFound): daemonsets.apps "nvidia-operator-validator" not found``
+
+The expected output should be:
+
+.. code-block:: text
+
+   Error from server (NotFound): daemonsets.apps "nvidia-operator-validator" not found
+   ...
+   Error from server (NotFound): daemonsets.apps "nvidia-operator-validator" not found
+   NAME                        DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR                                   AGE
+   nvidia-operator-validator   1         1         0       1            0           nvidia.com/gpu.deploy.operator-validator=true   17s
+
+Ensure the Validator succeeded
+""""""""""""""""""""""""""""""
+
+Next we'll need to wait for the Validator Pod to succeed:
+
+.. code-block:: bash
+
+  echo "Waiting for the NVIDIA Operator validations to complete..."
+  while ! sudo microk8s.kubectl logs -n gpu-operator-resources -l app=nvidia-operator-validator -c nvidia-operator-validator | grep "all validations are successful"
```

Comment:
Can we break this line into several? so the user can read it entirely without having to swipe

Reply:
Pushed a commit 6d795e7